app/src/main/java/io/gnosis/safe/ui/safe/SafeSelectionDialog.kt

```diff
@@ -0,0 +1,277 @@
+package io.gnosis.safe.ui.safe
+
+import android.content.Context
+import android.os.Bundle
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import androidx.fragment.app.FragmentActivity
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.Observer
+import androidx.lifecycle.liveData
+import androidx.navigation.Navigation
+import androidx.recyclerview.widget.DividerItemDecoration
+import androidx.recyclerview.widget.LinearLayoutManager
+import androidx.recyclerview.widget.RecyclerView
+import androidx.viewbinding.ViewBinding
+import io.gnosis.data.models.Safe
+import io.gnosis.safe.HeimdallApplication
+import io.gnosis.safe.R
+import io.gnosis.safe.databinding.DialogSafeSelectionBinding
+import io.gnosis.safe.databinding.ItemAddSafeBinding
+import io.gnosis.safe.databinding.ItemSafeBinding
+import io.gnosis.safe.di.Repositories
+import io.gnosis.safe.di.components.DaggerViewComponent
+import io.gnosis.safe.di.modules.ViewModule
+import io.gnosis.safe.ui.base.AppDispatchers
+import io.gnosis.safe.ui.base.BaseBottomSheetDialogFragment
+import io.gnosis.safe.ui.base.BaseStateViewModel
+import io.gnosis.safe.ui.base.UnsupportedViewType
+import io.gnosis.safe.utils.asMiddleEllipsized
+import pm.gnosis.svalinn.common.utils.visible
+import pm.gnosis.utils.asEthereumAddressString
+import java.lang.ref.WeakReference
+import javax.inject.Inject
+
+class SafeSelectionDialog : BaseBottomSheetDialogFragment<DialogSafeSelectionBinding>() {
+
+    @Inject
+    lateinit var viewModel: SafeSelectionViewModel
+
+    @Inject
+    lateinit var adapter: SafeSelectionAdapter
+
+    override fun inject() {
+        DaggerViewComponent.builder()
+            .viewModule(ViewModule(context!!))
+            .applicationComponent(HeimdallApplication[context!!])
+            .build()
```

Comment:
Building the view component can be pushed to the base class like it is for fragments.